File: challenge8.java
// Challenge 8
// Link to the problem: https://www.hackerrank.com/challenges/30-dictionaries-and-maps/problem

import java.util.*;
import java.io.*;

class Solution{
    public static void main(String []argh){
        Scanner in = new Scanner(System.in);
        String name, s;
        int phone;
        Map<String,Integer> telephoneBook = new HashMap<String,Integer>();
        int n = in.nextInt();
        for(int i = 0; i < n; i++)
        {
            name = in.next();
            phone = in.nextInt();
            telephoneBook.put(name,phone);
        }
        while(in.hasNext())
        {
            s = in.next();
            if(telephoneBook.get(s) == null)
                System.out.println("Not found");
            else
                System.out.println(s + "=" + telephoneBook.get(s));
        }
        in.close();
    }
}

File: challenge8.cpp
// Challenge 8
// Link to the problem: https://www.hackerrank.com/challenges/30-dictionaries-and-maps/problem

#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <algorithm>
#include <map>
#include <string>
using namespace std;


int main() {
    map<string,int> telephoneBook;
    int n, number;
    string name, s;
    
    cin >> n;
    
    for(int i = 0; i < n; i++)
    {
        cin >> name;
        cin >> number;
        telephoneBook.insert(pair<string,int>(name,number));
    }
    
    while(cin >> s)
    {
        //cin >> s;
        if(telephoneBook.find(s) == telephoneBook.end())
            cout << "Not found" << endl;
        else
            cout << s << "=" << (telephoneBook.find(s))->second << endl;
    }
    return 0;
}